### PR TITLE
Add fan levels auto-adjust

### PR DESCRIPTION
**Pull request by hungqbui**
G-Helper has been very useful since I made the switch from Armoury Crate. But tweaking with the fan levels have been annoying because they don't auto adjust. This is my first open source contribution ever so I'm sorry if my implementation is very elementary. Also, if there are any GitHub etiquettes that I'm not aware of please let me know.

### CODE REVIEW

File: app/Fans.cs [hunk 1]
```diff
@@ -450,6 +457,75 @@ private void ChartCPU_MouseMove(object? sender, MouseEventArgs e)
...
+            // Get the neighboring DataPoints of the hit point
+            DataPoint upperPoint = null;
+            DataPoint lowerPoint = null;
+
+            if (index > 0) 
+            {
+                lowerPoint = series.Points[index - 1];
+            }
+
+            if (index < series.Points.Count)
```
Comment (manciuszz) [inline app/Fans.cs:472]: Probably going to break the graph tooltips because of out of bounds exception when interacting with the last node in the graph.

**Comment by manciuszz** [comment]
I also see O(log n) potential over the current O(n) in-efficient code. However, that's just food for thought.

**Comment by hungqbui** [comment]
Yeah I definitely think there are way better ways to optimize it. The code itself isn't very DRY either. I'll try to figure it out, probably going to take me a while though. Thanks for the suggestion!

**Comment by manciuszz** [comment]
> Yeah I definitely think there are way better ways to optimize it. The code itself isn't very DRY either. I'll try to figure it out, probably going to take me a while though. Thanks for the suggestion!

Benchmark. Maybe play around with a Binary Search Tree. Benchmark again. I doubt there's going to be a significant difference considering we are dealing with a very low amount of graph nodes (points). However, if you never used any data structures before, here's your chance.

**Comment by hungqbui** [comment]
I haven't really learned Data Structures, at least not formally. But I'll look into using BST. I can't find a better implementation of this for an array.

**Comment by seerge** [comment]
@hungqbui hello, thanks for contribution, I will check it :)

**Comment by hungqbui** [comment]
> @hungqbui hello, thanks for contribution, I will check it :)

You're welcome. It is definitely not the best implementation though, but I haven't found anything better at the moment.

**Pull request merged**